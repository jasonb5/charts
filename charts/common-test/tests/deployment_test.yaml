suite: test deployment
values:
  - values/common.yaml
templates:
  - app.yaml
chart:
  appVersion: 0.1.0
tests:
  - it: should overwrite defaults
    set:
      workload: deployment

      minReadySeconds: 30
      progressDeadlineSeconds: 20
      replicas: 10
      revisionHistoryLimit: 20
      strategy:
        type: RollingUpdate
        rollingUpdate:
          maxSurge: 2
          maxUnavailable: 2
    asserts:
      - matchSnapshot:
          path: spec
        documentIndex: 0
      - hasDocuments:
          count: 2
  - it: should pass
    set:
      workload: deployment
    asserts:
      - matchSnapshot: {}
        documentIndex: 0
      - hasDocuments:
          count: 2
