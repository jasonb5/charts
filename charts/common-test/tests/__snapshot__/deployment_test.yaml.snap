should overwrite defaults:
  1: |
    minReadySeconds: 30
    progressDeadlineSeconds: 20
    replicas: 10
    revisionHistoryLimit: 20
    selector:
      matchLabels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: common-test
        common/workload: default
    strategy:
      rollingUpdate:
        maxSurge: 2
        maxUnavailable: 2
      type: RollingUpdate
    template:
      metadata:
        labels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: common-test
          common/workload: default
      spec:
        containers:
          - image: busybox:0.1.0
            name: common-test
should pass:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: common-test
        app.kubernetes.io/version: 0.1.0
        common/workload: default
        helm.sh/chart: common-test-0.1.0
      name: RELEASE-NAME-common-test
      namespace: NAMESPACE
    spec:
      replicas: 1
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: common-test
          common/workload: default
      strategy:
        type: Recreate
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: common-test
            common/workload: default
        spec:
          containers:
            - image: busybox:0.1.0
              name: common-test
